index.js

```diff
@@ -12,6 +14,11 @@ module.exports = (input, needle, replacement) => {
 	let ret = '';
 	let matchCount = 0;
 	let prevIndex = 0;
+	const startIndex = opts.fromIndex > 0 ? opts.fromIndex : 0;
+
+	if (startIndex > input.length) {
+		return input;
+	}
 
 	while (true) { // eslint-disable-line no-constant-condition
 		const index = input.indexOf(needle, prevIndex);
```

Comment:
- That if statement is for validating the fromIndex value, since a user may enter a value bigger than the input string length:
`t.is(m('foo', 'foo', 'bar', {fromIndex: -100}), 'bar')`
 
- IndexOf is using fromIndex value, as it can be seen on this test:
`t.is(m('foo foo foo foo foo', 'foo', 'bar', {fromIndex: 1}), 'foo bar bar bar bar');`

I added booth of these tests in my first commit and have been executed and tested on Travis.